framework/src/main.ts

```diff
@@ -42,7 +42,8 @@ export const run = async(
     progress?: Progress,
     language?: string,
     blacklist?: string[],
-    mode?: 'benchmarking'|'content-testing'|'load-testing'
+    mode?: 'benchmarking'|'content-testing'|'load-testing',
+    taskId?: string,
```

Comment:
this would need to be handed through to the executor and it's task builder

Reply:
I handed it through to the executor function and passed taskId to the BuildTaskList as the second parameter. Is it right?

Reply:
sounds very good so far